dask/array/overlap.py

```diff
@@ -534,7 +609,7 @@ def overlap(x, depth, boundary):
     new_chunks = tuple(
         ensure_minimum_chunksize(size, c) for size, c in zip(depths, x.chunks)
     )
-    x1 = x.rechunk(new_chunks)
+    x1 = x.rechunk(new_chunks)  # this is a no-op if x.chunks == new_chunks
 
     x2 = boundaries(x1, depth2, boundary2)
```

Comment:
Unfortunately `boundaries` gets pretty involved with slicing.  @gjoseph92 was planning to take a crack at `__getitem__`, but I wonder if there is a useful subset (perhaps constant or no boundary conditions) that we can do in the first pass.

Reply:
We could do constant boundaries, where the constant value is equal to, say, the median pixel intensity value (could be the median per-chunk, if that's easier to compute - it shouldn't require a separate pass through the whole array).

No boundary conditions: does this mean we just overlap where a neighbouring chunk exists? I've been fiddling with [something similar for other work](https://gist.github.com/GenevieveBuckley/decd23c22ee3417f7d78e87f791bc081), but this might cause problems when we get to the `trim` stage of `map_overlap`.

For the use case we have in mind, the most realistic boundary method will be "reflect". But "constant" boundaries will probably also be ok in a lot of cases.